src/nvim/quickfix.c

```diff
@@ -3983,17 +4000,54 @@ static int qf_buf_add_line(buf_T *buf, linenr_T lnum, const qfline_T *qfp,
   return OK;
 }
 
+static list_T *call_qftf_func(qf_list_T *qfl,
+                              int qf_winid,
+                              long start_idx,
+                              long end_idx)
```

Comment:
Vim uses `long` for the values passed to `dict_add_number` but casting to long is not required for `tv_dict_add_nr` in Neovim.

Reply:
Thanks, I will correct it.